skydance/protocol.py

```diff
@@ -408,11 +434,23 @@ class GetNumberOfZonesResponse(Response):
     # - Zone name
     # - Zone status (on/off)
 
+    def __init__(self, raw: bytes):
+        super().__init__(raw)
+        self._zones = []
+
+        for lbyte in self.cmd_data:
+            if (lbyte & DEVICE_BASE_TYPE_NORMAL) == DEVICE_BASE_TYPE_NORMAL:
+                zoneId = lbyte & 0x1f
+                self._zones.append(zoneId)
+
+
     @property
-    def number(self):
-        """Return number of zones available."""
-        return sum(1 for zone in self.body[12:28] if zone != 0)
+    def number(self) -> int:
```

Comment:
the docstring is important to generate docs properly

```suggestion
    def number(self) -> int:
    """Return number of zones available."""
```